src/rules/noUnnecessaryTypeAssertionRule.ts

```diff
@@ -44,13 +44,17 @@ export class Rule extends Lint.Rules.TypedRule {
         "This assertion is unnecessary since it does not change the type of the expression.";
 
     public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
+        const strictChecksEnabled = !!program.getCompilerOptions().strict;
```

Comment:
Love the variable names here. They're super informative!

One quick request - let's move the `program.getCompilerOptions()` into its own variable? Having it repeated a few times is relying on the assumption that the function is a quick lookup, and makes the code a little less readable IMO.

```suggestion
        const compilerOptions = program.getCompilerOptions();
        const strictChecksEnabled = !!compilerOptions.strict;
        // ...and the other two
```

Reply:
Fixed